src/server/command_registry.cc

```diff
@@ -130,15 +130,17 @@ bool CommandId::IsMultiTransactional() const {
   return CO::IsTransKind(name()) || CO::IsEvalKind(name());
 }
 
-uint64_t CommandId::Invoke(CmdArgList args, const CommandContext& cmd_cntx) const {
+uint64_t CommandId::Invoke(CmdArgList args, const CommandContext& cmd_cntx,
+                           std::optional<std::string_view> orig_cmd_name) const {
   int64_t before = absl::GetCurrentTimeNanos();
   handler_(args, cmd_cntx);
   int64_t after = absl::GetCurrentTimeNanos();
 
   ServerState* ss = ServerState::tlocal();  // Might have migrated thread, read after invocation
   int64_t execution_time_usec = (after - before) / 1000;
 
-  auto& ent = command_stats_[ss->thread_index()];
+  const std::string_view invoked_name = orig_cmd_name.value_or(name());
```

Comment:
Why don't you do this on the caller ? Then you can have this function accept a `string_view`, which is the command name that you will collect stats. That way you can get rid of the optional and that `value_or` thingy

Reply:
changed to do this in `InvokeCmd` and pass in `string_view` (although now the optional has to be handled by caller)